device/cuda/src/component_connection.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "edm/cell.hpp"
+#include "edm/cluster.hpp"
+#include "edm/measurement.hpp"
+
+#include "vecmem/memory/cuda/managed_memory_resource.hpp"
+#include "vecmem/memory/binary_page_memory_resource.hpp"
+#include "vecmem/containers/vector.hpp"
+
+#include "cuda/algorithms/component_connection.hpp"
+
+#include "details/sparse_ccl.cuh"
+
+namespace traccc::cuda {
+    host_measurement_collection
+    component_connection::operator()(
+        const host_cell_container & data
+    ) const {
+        vecmem::cuda::managed_memory_resource upstream;
+        vecmem::binary_page_memory_resource mem(upstream);
+
+        std::size_t total_cells = 0;
+
+        for (std::size_t i = 0; i < data.headers.size(); ++i) {
+            total_cells += data.items.at(i).size();
+        }
+
+        vecmem::vector<cell> cells(&mem);
+        cells.reserve(total_cells);
+
+        vecmem::vector<unsigned int> blocks(data.headers.size() + 1, &mem);
+        blocks[0] = 0;
+
+        vecmem::vector<float> out(4 * MAX_CLUSTERS_PER_MODULE * data.headers.size(), &mem);
```

Comment:
Probably I overlooked it, but where is `MAX_CLUSTERS_PER_MODULE` defined?

Reply:
It's defined in the header file!